invenio_records_rest/config.py

```diff
@@ -43,6 +44,7 @@ def _(x):
         pid_minter='recid',
         pid_fetcher='recid',
         search_class=RecordsSearch,
+        indexer=lambda: RecordIndexer(),
```

Comment:
Why the lambda? Is there problems if it would be just ``RecordIndexer()``?

Reply:
Yes, without it it complained that the application context was missing when building the app fixture for the tests because of this deepcopy: https://github.com/inveniosoftware/invenio-records-rest/blob/master/tests/conftest.py#L138
I realise that the copying was required for testing with different configurations though, so I'm not sure if the best thing would be to remove the lambda, pass directly the object instance and rewrite the tests.

Reply:
`invenio-indexer` CLI currently prevents any customization of the indexer class. It creates the default RecordIndexer class when doing a reindex.

Thus we have to make a choice:
1/ Not provide a way to override the RecordIndexer, only use the default class.
2/ Enable invenio-indexer CLI to use another class.

If we stay with the currently suggested solution developers will try to change the class and have a surprise when they do `invenio index reindex`.

Reply:
Ok, then we can put it back to ``indexer_class = RecordIndexer`` - sorry for the noise